### PR TITLE
Support AWS Partition deployment within Cloudformation.

### PR DESCRIPTION
**Pull request by gavinelder**
The following change moves to using AWS Supported [Pseudo paramaters](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html) where possible enabling deployment in AWS partitions `aws-cn` & `aws-govcloud`

* Allow passing of `AWS_PARTITION` into the dev docker 
* Change to using AWS Pseudo variables as part of the SDK & Cloud formation templating. 

* Deploy to `AWS-CN`
* Deploy to `AWS (West)`

### CODE REVIEW

File: deployments/auxiliary/cloudformation/panther-compliance-iam.yml [hunk 1]
```diff
@@ -119,25 +115,21 @@ Resources:
         Statement:
           - Effect: Allow
             Principal:
-              AWS: !Sub arn:aws:iam::${MasterAccountId}:root
+              AWS: !Sub arn:${AWS:: Partition}:iam::${MasterAccountId}:root
```
Comment (jacknagz) [inline deployments/auxiliary/cloudformation/panther-compliance-iam.yml]: Extra space here?

File: deployments/auxiliary/cloudformation/panther-compliance-iam.yml [hunk 2]
```diff
@@ -152,20 +144,19 @@ Resources:
...
       Description: The Panther master account assumes this role for automatic remediation of policy violations
-      MaxSessionDuration: 3600  # 1 hour
+      MaxSessionDuration: 3600 # 1 hour
       AssumeRolePolicyDocument:
         Version: 2012-10-17
         Statement:
           - Effect: Allow
             Principal:
-              AWS: !Sub arn:aws:iam::${MasterAccountId}:root
+              AWS: !Sub arn:${AWS:: Partition}:iam::${MasterAccountId}:root
```
Comment (jacknagz) [inline deployments/auxiliary/cloudformation/panther-compliance-iam.yml]: Here too

File: deployments/auxiliary/cloudformation/panther-log-processing-infra.yml [hunk 3]
```diff
@@ -80,13 +80,13 @@ Resources:
           - Sid: CloudTrailAclCheck
             Effect: Allow
             Principal:
-              Service: cloudtrail.amazonaws.com
+              Service: cloudtrail.${AWS::URLSuffix}
```
Comment (jacknagz) [inline deployments/auxiliary/cloudformation/panther-log-processing-infra.yml]: Super cool, never realized this was a variable!

Comment (jacknagz) [reply deployments/auxiliary/cloudformation/panther-log-processing-infra.yml]: You'll need to add `!Sub` in front of the string

Comment (jacknagz) [reply deployments/auxiliary/cloudformation/panther-log-processing-infra.yml]: And anywhere else this interpolation was added

File: deployments/auxiliary/cloudformation/panther-log-processing-infra.yml [hunk 4]
```diff
@@ -132,19 +132,17 @@ Resources:
...
         Version: 2012-10-17
         Statement:
-          -
-            # Reference: https://amzn.to/2ouFmhK
+          - # Reference: https://amzn.to/2ouFmhK
             Sid: S3NotificationPublish
             Effect: Allow
             Principal:
-              Service: 's3.amazonaws.com'
+              Service: 's3.${AWS::URLSuffix}'
```
Comment (jacknagz) [inline deployments/auxiliary/cloudformation/panther-log-processing-infra.yml]: Missing `!Sub`, and we can remove the single quotes

File: deployments/auxiliary/cloudformation/panther-log-processing-notifications.yml [hunk 5]
```diff
@@ -70,19 +76,19 @@ Resources:
           - Sid: AllowS3EventNotifications
             Effect: Allow
             Principal:
-              Service: s3.amazonaws.com
+              Service: s3.${AWS::URLSuffix}
```
Comment (jacknagz) [inline deployments/auxiliary/cloudformation/panther-log-processing-notifications.yml]: Add `!Sub`

File: deployments/auxiliary/cloudformation/panther-cloudwatch-events.yml [hunk 6]
```diff
@@ -109,18 +108,16 @@ Resources:
       PolicyDocument:
         Version: 2012-10-17
         Statement:
-          -
-            Sid: CloudWatchEventsPublish
+          - Sid: CloudWatchEventsPublish
```
Comment (austinbyers) [inline deployments/auxiliary/cloudformation/panther-cloudwatch-events.yml]: We usually like to keep the extra newline after the list element so they're more visually separated.

Are you using an auto-formatting tool that you'd recommend?

File: deployments/auxiliary/cloudformation/panther-cloudwatch-events.yml [hunk 7]
```diff
@@ -109,18 +108,16 @@ Resources:
...
             Effect: Allow
             Principal:
-              Service: events.amazonaws.com
+              Service: events.${AWS::URLSuffix}
```
Comment (austinbyers) [inline deployments/auxiliary/cloudformation/panther-cloudwatch-events.yml]: Neat!

File: deployments/auxiliary/cloudformation/panther-log-processing-notifications.yml (hunk 5 above)
Comment (austinbyers) [reply deployments/auxiliary/cloudformation/panther-log-processing-notifications.yml]: `cfn-lint` will catch this also, which is why CI is failing

File: dev.sh [hunk 8]
```diff
@@ -14,6 +14,7 @@ docker run \
     -e AWS_SESSION_TOKEN \
     -e AWS_SECURITY_TOKEN \
     -e AWS_REGION=$(if [ -z "$AWS_REGION" ]; then echo $AWS_DEFAULT_REGION; else echo $AWS_REGION; fi) \
+    -e AWS_PARTITION=$(if [ -z "$AWS_PARTITION" ]; then echo $AWS_PARTITION; else echo $AWS_PARTITION; fi) \
```
Comment (austinbyers) [inline dev.sh]: This can just be

`-e AWS_PARTITION`

File: deployments/auxiliary/cloudformation/panther-cloudwatch-events.yml (hunk 6 above)
Comment (jacknagz) [reply deployments/auxiliary/cloudformation/panther-cloudwatch-events.yml]: > Are you using an auto-formatting tool that you'd recommend?

Had the same thought 😂 we need this if so

**Comment by gavinelder** [comment]
@jacknagz @austinbyers I was not expecting any review so early thanks for the Feedback, 

With regards to the auto-formater it's https://prettier.io/  & https://github.com/mvdan/gofumpt . 

I will revert the Yaml changes as I feel it's going against your internal style guide.

**Comment by austinbyers** [comment]
@gavinelder if you are able to update this in all the other cloudformation files (and verify the deployment works for you in gov/cn), that'd be awesome

**Comment by austinbyers** [comment]
Merging this now before we make larger CFN changes, thanks for the contribution @gavinelder !

**Comment by gavinelder** [comment]
@austinbyers I have some small fix-ups for this .

**Comment by austinbyers** [comment]
I'm working on a branch which makes these changes to all of the CloudFormation (https://github.com/panther-labs/panther/compare/austin-mage-output), which should merge today or tomorrow, let me know if that addresses it

**Pull request merged**